server/command.go

```diff
@@ -147,31 +164,147 @@ func (p *Plugin) executeCommandQueue(args *model.CommandArgs) *model.CommandResp
+	oldPostID := split[2]
+	postToBeReQueued, _ := p.API.GetPost(oldPostID)
+	hashtagDateFormat, parsedMeetingMessage, errors := parseMeetingPost(meeting, postToBeReQueued)
+	if errors != nil {
+		return errors
+	}
+
+	originalPostDate := strings.ReplaceAll(strings.TrimSpace(parsedMeetingMessage.date), "_", " ") // reverse what we do to make it a valid hashtag
+	originalPostMessage := strings.TrimSpace(parsedMeetingMessage.textMessage)
+
+	today := time.Now()
+	local, _ := time.LoadLocation("Local")
+	formattedDate, _ := time.ParseInLocation(hashtagDateFormat, originalPostDate, local)
+	if formattedDate.Year() == 0 {
+		thisYear := today.Year()
+		formattedDate = formattedDate.AddDate(thisYear, 0, 0)
+	}
 
-	_, appErr = p.API.CreatePost(&model.Post{
+	if today.Year() <= formattedDate.Year() && today.YearDay() < formattedDate.YearDay() {
+		return responsef("We don't support re-queuing future items, only available for present and past items.")
+	}
+
+	hashtag, error := p.GenerateHashtag(args.ChannelId, false, -1, true, formattedDate.Weekday())
+	if error != nil {
+		return responsef("Error calculating hashtags. Check the meeting settings for this channel.")
+	}
+
+	numQueueItems, itemErr := calculateQueItemNumberAndUpdateOldItems(meeting, args, p, hashtag)
+	if itemErr != nil {
+		return itemErr
+	}
+
+	_, appErr := p.API.UpdatePost(&model.Post{
+		Id:        oldPostID,
 		UserId:    args.UserId,
 		ChannelId: args.ChannelId,
 		RootId:    args.RootId,
-		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems+1, message),
+		Message:   fmt.Sprintf("#### %v %v) %v", hashtag, numQueueItems, originalPostMessage),
 	})
 	if appErr != nil {
-		return responsef("Error creating post: %s", appErr.Message)
+		return responsef("Error updating post: %s", appErr.Message)
 	}
+	return &model.CommandResponse{Text: fmt.Sprintf("Item has been Re-queued to %v", hashtag), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+}
 
-	return &model.CommandResponse{}
+func parseMeetingPost(meeting *Meeting, post *model.Post) (string, ParsedMeetingMessage, *model.CommandResponse) {
+	var (
+		prefix            string
+		hashtagDateFormat string
+	)
+	if matchGroups := meetingDateFormatRegex.FindStringSubmatch(meeting.HashtagFormat); len(matchGroups) == 4 {
+		prefix = matchGroups[1]
+		hashtagDateFormat = strings.TrimSpace(matchGroups[2])
+	} else {
+		return "", ParsedMeetingMessage{}, responsef("Error 267")
+	}
+
+	var (
+		messageRegexFormat = regexp.MustCompile(fmt.Sprintf(`(?m)^#### #%s(?P<date>.*) ([0-9]+)\) (?P<message>.*)?$`, prefix))
+	)
+
+	matchGroups := messageRegexFormat.FindStringSubmatch(post.Message)
+	if len(matchGroups) == 4 {
+		parsedMeetingMessage := ParsedMeetingMessage{
+			date:        matchGroups[1],
+			number:      matchGroups[2],
+			textMessage: matchGroups[3],
+		}
+		return hashtagDateFormat, parsedMeetingMessage, nil
+	}
+	return hashtagDateFormat, ParsedMeetingMessage{}, responsef("Please ensure correct message format!")
```

Comment:
Under what circumstances will the user be receiving this message?

```suggestion
	return hashtagDateFormat, ParsedMeetingMessage{}, responsef("An error occurred processing the meeting hashtag.")
```

Reply:
If user tries to requeue non agenda message or message he/she has create with incorrect hashtag format.(we must remember user can always create agenda items manually given correct hashtag needs to be there)